spec/controllers/user_registrations_controller_spec.rb

```diff
@@ -3,21 +3,44 @@
 
 describe UserRegistrationsController, type: :controller do
 
+  before(:all) do
+    create(:mail_method)
+  end
+
   before do
     @request.env["devise.mapping"] = Devise.mappings[:spree_user]
   end
 
   describe "via ajax" do
     render_views
-    it "returns errors when registration fails" do
+
+    let(:user_params) do
+      {
+        email: "test@test.com",
+        password: "testy123",
+        password_confirmation: "testy123"
+      }
+    end
+
+    it "returns validation errors" do
       xhr :post, :create, spree_user: {}, :use_route => :spree
       expect(response.status).to eq(401)
       json = JSON.parse(response.body)
       expect(json).to eq({"email" => ["can't be blank"], "password" => ["can't be blank"]})
     end
 
+    it "returns error when emailing fails" do
+      allow(Spree::UserMailer).to receive(:confirmation_instructions).and_raise("Some error")
+
+      xhr :post, :create, spree_user: user_params, use_route: :spree
+
+      expect(response.status).to eq(401)
+      json = JSON.parse(response.body)
+      expect(json).to eq({"message" => "Something went wrong while creating your account. Check your email address and try again."})
```

Comment:
I'm not sure about this.

Upside: We can change text without changing the spec.
Downsides:
- The spec doesn't detect some translation mistakes:
  - Duplicate keys often lead to the selection of the wrong key.
  - A typo in the key (used in the view) may not be detected.
  - Mistakes in string interpolation can't be tested.
- The spec is harder to read and you don't know immediately what is displayed.

Should we bring that up in the forum and discuss if there are solutions to all of these problems?

Reply:
I tend to prefer using I18n.t so things can be reworded without changing the specs. Spelling mistakes in the view translation key would still be caught, no? Maybe the usefulness of either option depends on the situation, like with interpolated keys. 

Not being able to read the message whilst scanning the specs is a fair one, but my IDE displays the translation key automatically when viewing keys in specs, so I'm pretty ambivalent on that one...

Reply:
I also vote not to use `I18n.t` in the tests. :slightly_smiling_face: 

> Spelling mistakes in the view translation key would still be caught, no?

Hmm... @Matt-Yorkley Yes, you're right, it would catch a typo in the base key, although I think not mistakes in the scope.

Nice list of downsides, @mkllnk. An addition to that, if you have missing keys:

```ruby
expect("translation missing: en.missing_key").to eq("translation missing: en.missing_key")
```